source-mongodb/pull.go

```diff
@@ -316,15 +316,19 @@ func (c *capture) ChangeStream(ctx context.Context, client *mongo.Client, bindin
 	}
 
 	if err := cursor.Err(); err != nil {
-		return fmt.Errorf("cursor error for change stream %s: %w", res.Collection, err)
+		return fmt.Errorf("cursor error for change stream: %w", err)
 	}
 
 	defer cursor.Close(ctx)
 
 	return nil
 }
 
-const BackfillBatchSize = 1024
+// We use a very large batch size here, however the go mongo driver will
+// automatically use a smaller batch size that prevents memory usage to blow out
+// of proportion. The large number here basically is a way for us to ask the
+// driver to choose a reasonable batch size for us.
+const BackfillBatchSize = 262144
```

Comment:
We are calculating when to checkpoint off of having emitted this many documents, and a single checkpoint should not have this many documents. We're using 50k for SQL captures as their default backfill chunk size so that might be a good choice here too, see https://estuaryworkspace.slack.com/archives/C03Q2NRFKDL/p1692036679355679